Yoast/wordpress-seo
# P3-139 Don't do certain things when the WordPress environment type is not production

## Context


* We want to avoid to run some code when the WordPress environment type is not "production" so that some features run only when necessary: on live, production, sites.

## Summary


This PR can be summarized in the following changelog entry:

* Avoids to run unnecessary features on sites that aren't production sites.

## Relevant technical choices:


## Test instructions

This PR can be tested by following these steps:

**Test instructions for the Ryte Site Health part:** 

- on trunk
- in `get_remote()` in `admin/ryte/class-ryte-request.php` add this on the first line of the function: `$target_url = "https://yoast.com";` so that the Ryte Site Health check runs against a reachable site
- go to the WordPress admin > Tools > Site Health 
- expand the "Passed tests" section 
- see the Ryte check that "Your site can be found by search engines"
- switch to this branch and build 
- add `define( 'WP_ENVIRONMENT_TYPE', 'staging' );` in your `wp-config.php` file (note: on the Docker environment the file is `config/basic-wordpress-config.php`)
- refresh the Site Health page 
- observe the Ryte check didn't run

**Test instructions for sitemap and tracking**
Due to the nature of these two pieces of code, testing is limited.

Sitemaps
* Ensure tests pass for the `class-sitemap-admin.php`
* Create a new post and publish.
* Run `./wp.sh cron event list` on your Docker setup. See that the `wpseo_ping_search_engines` was added (check timestamp as well for comparison)
* Add `define( 'WP_ENVIRONMENT_TYPE', 'development' );` to your `basic-wordpress-config.php` file.
* Create another new post.
* Run the CLI command again.
* Notice that no new cron was added (based on the current timestamp)

Tracking
* Ensure that tests pass.

## UI changes

## Documentation

## Quality assurance


Fixes [P3-139]
